### ISSUE TITLE
Library cannot be built with tools from the 14.3 branch

### ISSUE THREAD

**Issue by samchesney** [xmos/lib_mic_array#43]
The issue appears to be the use of `null` in decimator_interface.xc.

Seen when building AN00217_app_high_resolution_delay_example:
```xc
Checking build modules
Using build modules: lib_mic_array(3.0.0) lib_xassert(3.0.0) lib_logging(2.1.0) lib_dsp(3.0.0)
Analyzing debug_printf.c
Analyzing dsp_adaptive.c
Analyzing dsp_design.c
Analyzing dsp_filters.c
Analyzing dsp_math.c
Analyzing dsp_matrix.c
Analyzing dsp_statistics.c
Analyzing dsp_tables.c
Analyzing dsp_vector.c
Analyzing dsp_ds3.c
Analyzing dsp_os3.c
Analyzing app_high_resolution_delay_example.xc
Analyzing decimator_interface.xc
/.../lib_mic_array/lib_mic_array/src/decimator_interface.xc:16:34: error: types of conditional operator result do not match
        ch[0] = isnull(a) ? null : (unsigned)a;
                            ~~~~~^~~~~~~~~~~~~
/.../lib_mic_array/lib_mic_array/src/decimator_interface.xc:17:34: error: types of conditional operator result do not match
        ch[1] = isnull(b) ? null : (unsigned)b;
                            ~~~~~^~~~~~~~~~~~~
/.../lib_mic_array/lib_mic_array/src/decimator_interface.xc:18:34: error: types of conditional operator result do not match
        ch[2] = isnull(c) ? null : (unsigned)c;
                            ~~~~~^~~~~~~~~~~~~
/.../lib_mic_array/lib_mic_array/src/decimator_interface.xc:19:34: error: types of conditional operator result do not match
        ch[3] = isnull(d) ? null : (unsigned)d;
                            ~~~~~^~~~~~~~~~~~~
xmake[1]: *** [.build/_l_mic_array/src/decimator_interface.xc.pca.xml.decouple] Error 1
xmake: *** [analyze] Error 2
ERROR: Failed to build: ./examples/AN00217_app_high_resolution_delay_example (Default)
```

**Comment by robert-lytton** [comment]
I think you wanted an 'unsigned' there such as '0'.
Or at least an invalid 'streaming chanend' id... is '0' valid?... no, so your OK (it would be valid if it were a tileref)
Or you could cast the internal representation of 'null' to an 'unsigned' to see what happens ... what kind of null is it.... :-)

**Comment by samchesney** [comment]
Fixed in commit 66f7e22546fd9dfaf6484007d7ea29681d6c9240
